From 2a3224cd19447f6cfd1f3f018edc11a673337072 Mon Sep 17 00:00:00 2001
From: Noam Lessner <nlessner@habana.ai>
Date: Mon, 03 Feb 2020 12:21:22 +0200
Subject: [PATCH] [SW-8229] goya2: config rr for pcie with spi

When spi controller is enabled with strap also configure
PCIe range-register to enable access to controller over
PCIe.
Replace all consts with defines for easier understanding

Change-Id: I85936fe341db144723faf4529efcfa4356bd5a63
Signed-off-by: Noam Lessner <nlessner@habana.ai>
---

diff --git a/common/pcie.c b/common/pcie.c
index d6c19db..e888c05 100644
--- a/common/pcie.c
+++ b/common/pcie.c
@@ -74,17 +74,21 @@
 
 void pcie_recovery(void)
 {
+	uint8_t spi_en = arcboot_rom_global.config_fuse &
+			EFUSE_SYSCFG_SPI_CONTROLLER;
+
 	security_debug_if_secure();
 	security_debug_if_shared();
 
-	if (arcboot_rom_global.config_fuse & EFUSE_SYSCFG_DISABLE_DEBUG_IF_MASK)
-		secuirty_debug_if_hbw_config();
-	else
+	if ((arcboot_rom_global.config_fuse &
+	     EFUSE_SYSCFG_DISABLE_DEBUG_IF_MASK) == 0) {
+		secuirty_debug_if_rr_hbw_config(spi_en);
 		secuirty_i2c_s_enable();
+	}
 
 	security_spi_dma_mmu_bypass();
 	security_mem_cpy_mmu_bypass();
-	security_pcie_config();
+	security_pcie_rr_config(spi_en);
 	pcie_config();
 }
 
diff --git a/common/security.c b/common/security.c
index 62e5c64..081bfaa 100644
--- a/common/security.c
+++ b/common/security.c
@@ -14,45 +14,45 @@
 static void security_disable_i2c_s_mstr_if(void)
 {
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(0x1,
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		      mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
 static void security_disable_smi_mstr_if(void)
 {
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(0x1,
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		      mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
 static void security_disable_jtag_mstr_if(void)
 {
 	/* override AXI prot bit by master IF to secure, unprivlaged access */
-	regmap_writel(0x1,
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
 		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		      mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 }
 
@@ -66,18 +66,25 @@
 void security_debug_if_secure(void)
 {
 	/* Configure SMI,I2C and JTAG to be "secure" */
-	regmap_writel(0x1, mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	regmap_writel(0x1, mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
-	regmap_writel(0x1, mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
+		      mmI2C_S_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
+		      mmSMI_MSTR_IF_CORE_HBW_AXPROT_OVRD);
+	regmap_writel(MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN,
+		      mmJT_MSTR_IF_CORE_HBW_AXPROT_OVRD);
 }
 
 void security_debug_if_shared(void)
 {
 	/* Configure SMI,I2C and JTAG to be "shared" engine */
-	regmap_writel(0x400, mmI2C_S_MSTR_IF_AXUSER_HB_SEC);
-	regmap_writel(0x400, mmSMI_MSTR_IF_AXUSER_HB_SEC);
-	regmap_writel(0x400, mmJT_MSTR_IF_AXUSER_HB_SEC);
-	regmap_writel(0x1, mmPSOC_MSTR_IF_AXUSER_CORE);
+	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
+		      mmI2C_S_MSTR_IF_AXUSER_HB_SEC);
+	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
+		      mmSMI_MSTR_IF_AXUSER_HB_SEC);
+	regmap_writel(MSTR_IF_AXUSER_HB_SEC_MMBP,
+		      mmJT_MSTR_IF_AXUSER_HB_SEC);
+	regmap_writel(MSTR_IF_AXUSER_CORE_SHARED,
+		      mmPSOC_MSTR_IF_AXUSER_CORE);
 }
 
 uint32_t security_key_soc_location_offset(uint32_t hash_key_index)
@@ -107,230 +114,294 @@
 
 void security_spi_dma_mmu_bypass(void)
 {
-	regmap_writel(mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER,0x400);
+	regmap_writel(mmPSOC_GLOBAL_CONF_SPI_DMA_AWUSER,
+		      MSTR_IF_AXUSER_HB_SEC_MMBP);
 }
 
 void security_mem_cpy_mmu_bypass(void)
 {
-	regmap_writel(mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER,0x400);
+	regmap_writel(mmPSOC_GLOBAL_CONF_MEM_CPY_AXUSER,
+		      MSTR_IF_AXUSER_HB_SEC_MMBP);
 }
 
-static void secuirty_elbi_lbw_config(void)
+static void secuirty_elbi_rr_lbw_config(uint8_t spi_controller_en)
 {
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_LBW,
 		mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
-	regmap_writel(0x0460000,
-		mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-	regmap_writel(0x0461000,
-		mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-	regmap_writel(0x3ffffff,
-		mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+
+	if (spi_controller_en) {
+		regmap_writel(SECURITY_QSPI_START_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
+		regmap_writel(SECURITY_QSPI_END_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
+		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_2);
+		regmap_writel(SECURITY_PSOC_END_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_2);
+	} else {
+		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
+		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
+		regmap_writel(SECURITY_PSOC_END_LBW,
+			mmPCIE_ELBI_RR_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+	}
 }
 
-static void secuirty_dcon_lbw_config(void)
+static void secuirty_dcon_rr_lbw_config(uint8_t spi_controller_en)
 {
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_LBW,
 		mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_0);
-	regmap_writel(0x0460000,
-		mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
-	regmap_writel(0x0461000,
-		mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
-	regmap_writel(0x3ffffff,
-		mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+
+	if (spi_controller_en) {
+		regmap_writel(SECURITY_QSPI_START_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
+		regmap_writel(SECURITY_QSPI_END_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
+		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_2);
+		regmap_writel(SECURITY_PSOC_END_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_2);
+	} else {
+		regmap_writel(SECURITY_PSOC_SECURITY_START_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_0);
+		regmap_writel(SECURITY_PSOC_SECURITY_END_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MIN_1);
+		regmap_writel(SECURITY_PSOC_END_LBW,
+		       mmDCON0_LBW_RTR_IF_MSTR_IF_RR_SHRD_LBW_PRIV_RANGE_MAX_1);
+	}
 }
 
-static void secuirty_dcon_if0_hbw_config(void)
+static void secuirty_dcon_if0_rr_hbw_config(uint8_t spi_controller_en)
 {
-	/* RR for pcie dcon hbw 0x0 - 0x00_07FF_A000 */
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0x7FFA000,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0x0,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 
-	/* RR for pcie dcon hbw 0x00_7FF_B000 -  0x3F_FFFFF_FFFF */
-	regmap_writel(0x7FFB000,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-	regmap_writel(0x0,
-		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	regmap_writel(0xFFFFFFFF,
+	if (spi_controller_en) {
+		regmap_writel(SECURITY_QSPI_START_HBW,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+		regmap_writel(SECURITY_QSPI_END_HBW,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	} else {
+		regmap_writel(SECURITY_FLASH_START_HBW,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+		regmap_writel(SECURITY_FLASH_END_HBW,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	}
+
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+		mmDCON0_HBW_RTR_IF0_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
 }
 
-static void secuirty_dcon_if1_hbw_config(void)
+static void secuirty_dcon_if1_rr_hbw_config(uint8_t spi_controller_en)
 {
-	/* RR for pcie dcon hbw 0x0 - 0x00_07FF_A000 */
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0x0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
-	regmap_writel(0x7FFA000,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-	regmap_writel(0x0,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
 
-	/* RR for pcie dcon hbw 0x00_7FF_B000 -  0x3F_FFFFF_FFFF */
-	regmap_writel(0x7FFB000,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-	regmap_writel(0x0,
-		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
-	regmap_writel(0xFFFFFFFF,
+	if (spi_controller_en) {
+		regmap_writel(SECURITY_QSPI_START_HBW,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+		regmap_writel(SECURITY_QSPI_END_HBW,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	} else {
+		regmap_writel(SECURITY_FLASH_START_HBW,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
+		regmap_writel(SECURITY_FLASH_END_HBW,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
+		   mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
+	}
+
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
+		mmDCON0_HBW_RTR_IF1_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
 }
 
-static void secuirty_i2c_s_hbw_config(uint8_t spi_controller_en)
+static void secuirty_i2c_s_rr_hbw_config(uint8_t spi_controller_en)
 {
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
 
 	if (spi_controller_en) {
-		regmap_writel(0x7FFC444,
+		regmap_writel(SECURITY_QSPI_START_HBW,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFC445,
+		regmap_writel(SECURITY_QSPI_END_HBW,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	} else {
-		regmap_writel(0x7FFA000,
+		regmap_writel(SECURITY_FLASH_START_HBW,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFB000,
+		regmap_writel(SECURITY_FLASH_END_HBW,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	}
 
-	regmap_writel(0x7FFC460,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(0x7FFC461,
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		mmI2C_S_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
 }
 
-static void secuirty_smi_hbw_config(uint8_t spi_controller_en)
+static void secuirty_smi_rr_hbw_config(uint8_t spi_controller_en)
 {
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
 
 	if (spi_controller_en) {
-		regmap_writel(0x7FFC444,
+		regmap_writel(SECURITY_QSPI_START_HBW,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFC445,
+		regmap_writel(SECURITY_QSPI_END_HBW,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	} else {
-		regmap_writel(0x7FFA000,
+		regmap_writel(SECURITY_FLASH_START_HBW,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFB000,
+		regmap_writel(SECURITY_FLASH_END_HBW,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	}
 
-	regmap_writel(0x7FFC460,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(0x7FFC461,
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		mmSMI_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
 }
 
-static void secuirty_jtag_hbw_config(uint8_t spi_controller_en)
+static void secuirty_jtag_rr_hbw_config(uint8_t spi_controller_en)
 {
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_LSB,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_0);
-	regmap_writel(0,
+	regmap_writel(SECURITY_PSOC_START_HBW_MSB,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_0);
 
 	if (spi_controller_en) {
-		regmap_writel(0x7FFC444,
+		regmap_writel(SECURITY_QSPI_START_HBW,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFC445,
+		regmap_writel(SECURITY_QSPI_END_HBW,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	} else {
-		regmap_writel(0x7FFA000,
+		regmap_writel(SECURITY_FLASH_START_HBW,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_0);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_0);
-		regmap_writel(0x7FFB000,
+		regmap_writel(SECURITY_FLASH_END_HBW,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_1);
-		regmap_writel(0,
+		regmap_writel(SECURITY_HBW_MSB_ZERO,
 			mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_1);
 	}
 
-	regmap_writel(0x7FFC460,
+	regmap_writel(SECURITY_PSOC_SECURITY_START_HBW,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_1);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_1);
-	regmap_writel(0x7FFC461,
+	regmap_writel(SECURITY_PSOC_SECURITY_END_HBW,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_LO_2);
-	regmap_writel(0,
+	regmap_writel(SECURITY_HBW_MSB_ZERO,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MIN_HI_2);
-	regmap_writel(0xFFFFFFFF,
+	regmap_writel(SECURITY_PSOC_END_HBW_LSB,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_LO_2);
-	regmap_writel(0x3F,
+	regmap_writel(SECURITY_PSOC_END_HBW_MSB,
 		mmJT_MSTR_IF_RR_SHRD_HBW_PRIV_RANGE_MAX_HI_2);
 }
 
-void security_pcie_config(void)
+void security_pcie_rr_config(uint8_t spi_en)
 {
 	regmap_writel(PCIE_WRAP_PCIE_PROT_OVR_WR_EN_MASK |
 		      PCIE_WRAP_PCIE_PROT_OVR_RD_EN_MASK,
 		      mmPCIE_WRAP_PCIE_PROT_OVR);
 	regmap_writel(PCIE_MSTR_RR_MSTR_IF_AXUSER_CORE_SHARED_MASK,
 		      mmPCIE_MSTR_RR_MSTR_IF_AXUSER_CORE);
-	secuirty_elbi_lbw_config();
-	secuirty_dcon_lbw_config();
-	secuirty_dcon_if0_hbw_config();
-	secuirty_dcon_if1_hbw_config();
+	secuirty_elbi_rr_lbw_config(spi_en);
+	secuirty_dcon_rr_lbw_config(spi_en);
+	secuirty_dcon_if0_rr_hbw_config(spi_en);
+	secuirty_dcon_if1_rr_hbw_config(spi_en);
 }
 
 /* If DBG I/F were not disabled, program the RR to allow access
  * only to SPI flash and PSOC_SECURITY
  */
-void secuirty_debug_if_hbw_config(void)
+void secuirty_debug_if_rr_hbw_config(uint8_t spi_en)
 {
-	uint8_t spi_en = arcboot_rom_global.config_fuse &
-				EFUSE_SYSCFG_SPI_CONTROLLER;
-	secuirty_i2c_s_hbw_config(spi_en);
-	secuirty_smi_hbw_config(spi_en);
-	secuirty_jtag_hbw_config(spi_en);
+	secuirty_i2c_s_rr_hbw_config(spi_en);
+	secuirty_smi_rr_hbw_config(spi_en);
+	secuirty_jtag_rr_hbw_config(spi_en);
 }
 
 void secuirty_i2c_s_enable(void)
diff --git a/common/security.h b/common/security.h
index 4e63159..863968e 100644
--- a/common/security.h
+++ b/common/security.h
@@ -36,8 +36,47 @@
 
 #define SECURITY_KEY_HASH_ARRAY_SIZE		(SECURITY_DWORD_PER_KEY * \
 						SECURITY_KEY_HASH_COUNT)
-#define SECURITY_KEY_HASH_ARRAY_BIT_SIZE	(SECURITY_KEY_HASH_ARRAY_SIZE *\
-						sizeof(uint32_t))
+#define SECURITY_KEY_HASH_ARRAY_BIT_SIZE	(SECURITY_KEY_HASH_ARRAY_SIZE \
+						* sizeof(uint32_t))
+
+#define SECURITY_LBW_MASK			0xFFFFFF
+
+#define SECURITY_PSOC_START_LBW			0x0
+#define SECURITY_QSPI_START_LBW			(mmPSOC_QSPI_BASE & \
+						 SECURITY_LBW_MASK)
+#define SECURITY_QSPI_END_LBW			(mmPSOC_UART_0_BASE & \
+						 SECURITY_LBW_MASK)
+
+#define SECURITY_PSOC_SECURITY_START_LBW	(mmPSOC_SECURITY_BASE & \
+						 SECURITY_LBW_MASK)
+#define SECURITY_PSOC_SECURITY_END_LBW	(mmJT_MSTR_IF_RR_SHRD_HBW_BASE & \
+					 SECURITY_LBW_MASK)
+#define SECURITY_PSOC_END_LBW			0x3FFFFFF
+
+#define SECURITY_PSOC_START_HBW_MSB		0x0
+#define SECURITY_PSOC_START_HBW_LSB		0x0
+
+#define SECURITY_QSPI_START_HBW			(mmPSOC_QSPI_BASE >> 24)
+#define SECURITY_QSPI_END_HBW			(mmPSOC_UART_0_BASE >> 24)
+#define SECURITY_PSOC_SECURITY_START_HBW	(mmPSOC_SECURITY_BASE >> 24)
+#define SECURITY_PSOC_SECURITY_END_HBW		(mmJT_MSTR_IF_RR_SHRD_HBW_BASE \
+						 >> 24)
+#define SECURITY_FLASH_START_HBW		0x7FFA000
+#define SECURITY_FLASH_END_HBW			0x7FFB000
+
+#define SECURITY_HBW_MSB_ZERO			0x0
+
+#define SECURITY_PSOC_END_HBW_MSB		0x3F
+#define SECURITY_PSOC_END_HBW_LSB		0xFFFFFFFF
+
+#define SECURITY_MAX_FW_SEC_VERSION		128
+#define SECURITY_FW_VER_LAST_DWORD		3
+#define SECURITY_FW_VER_LAST_BYTE		31
+
+#define MSTR_IF_AXUSER_HB_SEC_MMBP		0x400
+#define MSTR_IF_CORE_HBW_AXPROT_OVRD_OVRD_EN	1
+#define MSTR_IF_AXUSER_CORE_SHARED		1
+
 
 void security_disable_debug_if(void);
 void security_debug_if_secure(void);
@@ -45,8 +84,8 @@
 uint32_t security_key_soc_location_offset(uint32_t hash_key_index);
 void security_spi_dma_mmu_bypass(void);
 void security_mem_cpy_mmu_bypass(void);
-void security_pcie_config(void);
-void secuirty_debug_if_hbw_config(void);
+void security_pcie_rr_config(uint8_t spi_en);
+void secuirty_debug_if_rr_hbw_config(uint8_t spi_en);
 void secuirty_i2c_s_enable(void);
 uint8_t security_find_min_fw_version(uint32_t *fw_version);
 int security_get_key_hash(uint32_t *hash, uint8_t customer);
